.github/scripts/ext-debian.sh

```diff
@@ -68,108 +98,123 @@ build_bin_package() {
+
+      ln "../$SUITE/${debian_package_name}_${version}.dsc" .
+      ln "../$SUITE/${debian_package_name}_${short_version}.orig.tar.gz" .
+
+      if [ -f "../$SUITE/${debian_package_name}_${version}.debian.tar.xz" ]; then
+        ln "../$SUITE/${debian_package_name}_${version}.debian.tar.xz" .
+      fi
+      if [ -f "../$SUITE/${debian_package_name}_${version}.diff.gz" ]; then
+        ln "../$SUITE/${debian_package_name}_${version}.diff.gz" .
+      fi
+
+      cd - >/dev/null 2>&1
+    fi
+
+    echo "Publishing source package $debian_package_name into $PKG_PUBLISH_PATH"
   fi
 
-  DEB_CONTROL_FILE="$PKG_BUILD_DIR/$PACKAGE_NAME/debian/control"
+  DEB_CONTROL_FILE="$PKG_BUILD_PATH/$PACKAGE_NAME/debian/control"
   ALL_ARCH="$ARCH,all"
 
   for target_arch in $(echo $ALL_ARCH | sed "s/,/ /g"); do
-      cat "$DEB_CONTROL_FILE" | grep ^Package: | cut -d' ' -f2 | while read -r bin_pkg; do
-          DEB_BIN_PKG_PATH="$(pwd)/${bin_pkg}_${version}_${target_arch}.deb"
-          if [ -f "$DEB_BIN_PKG_PATH" ]; then
-              if [ "$allow_failing_bin_package" == "true" ]; then                
-                # If the source package/version already exists, allow the bin package build to fail (already exists)
-                reprepro --basedir "$PKG_REPO_PATH" includedeb "$CODENAME" "$DEB_BIN_PKG_PATH" || true
-                echo "Ingested binary package $DEB_BIN_PKG_PATH into $PKG_REPO_PATH"
-              else
-                reprepro --basedir "$PKG_REPO_PATH" includedeb "$CODENAME" "$DEB_BIN_PKG_PATH"
-                echo "Ingested binary package $DEB_BIN_PKG_PATH into $PKG_REPO_PATH"
-              fi
-              echo "CHLOG:Published ${bin_pkg}_${version}_${target_arch}.deb in $STAGE $DISTRO $CODENAME $ARCH from $PKG_LINE"
-          else
-              echo "Package $bin_pkg does not exist for $target_arch"
-          fi
-      done
+    cat "$DEB_CONTROL_FILE" | grep ^Package: | cut -d' ' -f2 | while read -r bin_pkg; do
+      DEB_BIN_PKG_PATH="$(pwd)/${bin_pkg}_${version}_${target_arch}.deb"
+
+      if [ -f "$DEB_BIN_PKG_PATH" ]; then
+        mkdir -p $PKG_PUBLISH_PATH/$DISTRO/$CODENAME/$SUITE
+        cp "$DEB_BIN_PKG_PATH" "$PKG_PUBLISH_PATH/$DISTRO/$CODENAME/$SUITE"
+
+        if [ "$LOCAL_BUILD" == "false" ] && [ "$SUITE" == "stable" ]; then
+          mkdir -p "$PKG_PUBLISH_PATH/$DISTRO/$CODENAME/$COMPONENT"
+          cd "$PKG_PUBLISH_PATH/$DISTRO/$CODENAME/$COMPONENT" >/dev/null 2>&1
+          ln "../$SUITE/${bin_pkg}_${version}_${target_arch}.deb" .
+          cd - >/dev/null 2>&1
+        fi
+
+        echo "Publishing deb package(s) $bin_pkg into $PKG_PUBLISH_PATH"
+        echo "CHLOG:Published ${bin_pkg}_${version}_${target_arch}.deb in $DISTRO/$CODENAME/$STAGE from $PKG_LINE"
+      else
+        echo "Package $bin_pkg does not exist for $target_arch"
+      fi
+    done
   done
+}
+
+archive_setup_scripts() {
+  # Following allows for internal dependencies
```

Comment:
question: can you elaborate on this?  What would qualify as an internal dependency?

Reply:
This part is equivalent of [this](https://github.com/regolith-linux/voulage/blob/f4967fe077f94d3af36dfee7efd085e8770d4894/.github/workflows/build-deb-v4.yml#L111-L121) which I moved from action file to here.

Not entirely sure if it is actually needed or not, but I think if a package that's being built relies on other packages that are maintained by Regolith (and basically only exists in `archive.regolith-desktop.com` and not Ubuntu or Debian public repos)

Reply:
It's actually needed :smile: .  Presently we have one such case: `sway-regolith` depends on `trawldb`.  My current approach is to break a build that doesn't already have some version of `trawldb` in the repo (for example, a new OS-codename being added) which result in commits like [this](https://github.com/regolith-linux/voulage/commit/7530d6fe07ec2dddb4ed153626e92de6cb935e5a) and then a revert.  Would be really nice to not have to deal with that sort of thing.

Reply:
That is a great point! I've also thinking about this and I think I might have a way forward to decentralize the process of building and publishing packages. If that works, then dependency management will ultimately boil down to "run the GitHub Action of the dependant package first and then run the Action of the other package."

We can track this further in #45 separately after getting this merged and get the ball rolling for unified archives in new server.